sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -255,6 +257,60 @@ shutdown_error_redact_t shutdown_redact_policy(void) {
   return shutdown_redact_policy_inline(raw_state);
 }
 
+enum {
+  /**
+   * Length of the error message prefix.
+   */
+  kMsgPrefixLen = 14,
+  /**
+   * Length of the hexadecimal representation.
+   */
+  kHexCharsLen = 8,
+  /**
+   * Base address of UART.
+   */
+  kUartBase = TOP_EARLGREY_UART0_BASE_ADDR,
+};
+
+static_assert((kMsgPrefixLen + kHexCharsLen) <= 32,
+              "Total message length must be less than TX FIFO size.");
+
+/**
+ * Prints a fixed-length message prefix.
+ *
+ * @param prefix Prefix to print.
+ */
+static ALWAYS_INLINE void shutdown_print_prefix(const char *prefix) {
```

Comment:
Would it make more sense for this to take a C string and exit if you hit the maximum prefix length? Since you call this and print32 together, maybe it makes sense to merge them and have the `: 0x` be filled in automatically.

If there's a good reason not to do this, it should be documented.